appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -177,46 +177,31 @@ const (
 	EnvironmentReasonErrorOccurred                   = "ErrorOccurred"
 )
 
-// Update Status.Condition field of Environment
+// Update .status.conditions field of Environment
 func updateStatusConditionOfEnvironment(ctx context.Context, client client.Client, message string,
 	environment *appstudioshared.Environment, conditionType string,
 	status metav1.ConditionStatus, reason string, log logr.Logger) error {
-	// Check if condition with same type is already set, if Yes then check if content is same,
-	// If content is not same update LastTransitionTime
-	index := -1
-	for i, Condition := range environment.Status.Conditions {
-		if Condition.Type == conditionType {
-			index = i
-			break
-		}
+
+	newCondition := metav1.Condition{
+		Type:    conditionType,
+		Message: message,
+		Status:  status,
+		Reason:  reason,
 	}
 
-	now := metav1.Now()
+	changed, newConditions := insertOrUpdateConditionsInSlice(newCondition, environment.Status.Conditions)
 
-	if index == -1 {
-		environment.Status.Conditions = append(environment.Status.Conditions,
-			metav1.Condition{
-				Type:               conditionType,
-				Message:            message,
-				LastTransitionTime: now,
-				Status:             status,
-				Reason:             reason,
-			})
-	} else if environment.Status.Conditions[index].Message != message ||
-		environment.Status.Conditions[index].Reason != reason ||
-		environment.Status.Conditions[index].Status != status {
-		environment.Status.Conditions[index].Reason = reason
-		environment.Status.Conditions[index].Message = message
-		environment.Status.Conditions[index].Status = status
-		environment.Status.Conditions[index].LastTransitionTime = now
-	}
+	if changed {
```

Comment:
Just a clarification question:  So before, we were always making the client.Status().Update call regardless if the bindingCondition changed or not. (Hence the high put request rate).  At this point, we will only call it if it only changed....right?

Reply:
Yup, exactly.